plugin/trino-hive/src/main/java/io/trino/plugin/hive/authentication/ImpersonatingHdfsAuthentication.java

```diff
@@ -33,10 +34,10 @@ public ImpersonatingHdfsAuthentication(@ForHdfs HadoopAuthentication hadoopAuthe
     }
 
     @Override
-    public <R, E extends Exception> R doAs(String user, GenericExceptionAction<R, E> action)
+    public <R, E extends Exception> R doAs(ConnectorIdentity connectorIdentity, GenericExceptionAction<R, E> action)
             throws E
     {
-        return executeActionInDoAs(createProxyUser(user), action);
+        return executeActionInDoAs(createProxyUser(connectorIdentity.getUser()), action);
```

Comment:
You should provide a bean that handles how `identity` is transformed to `user`, so it could be customized.